### PR TITLE
Mocking updates

### PR DESCRIPTION
**Pull request by nedwors**
Extracts methods for faking/testing into their own class

### CODE REVIEW

File: src/Providers/SoapServiceProvider.php [hunk 1]
```diff
@@ -8,13 +8,14 @@
...
 use RicorocksDigitalAgency\Soap\Soap;
+use RicorocksDigitalAgency\Soap\Support\Fakery\Fakery;
 
 class SoapServiceProvider extends ServiceProvider
 {
 
     public function register()
     {
-        $this->app->singleton('soap', fn() => app(Soap::class));
+        $this->app->singleton('soap', fn() => app(Soap::class, ['fakery' => app(Fakery::class)]));
```
Comment (lukeraymonddowning) [inline src/Providers/SoapServiceProvider.php:18]: We shouldn't need the call to `app(Fakery::class)`

File: src/Soap.php [hunk 2]
```diff
@@ -1,48 +1,27 @@
...
-
-use Closure;
-use Illuminate\Support\Str;
-use PHPUnit\Framework\Assert as PHPUnit;
 use RicorocksDigitalAgency\Soap\Parameters\Node;
 use RicorocksDigitalAgency\Soap\Request\Request;
-use RicorocksDigitalAgency\Soap\Response\Response;
+use RicorocksDigitalAgency\Soap\Support\Fakery\Fakery;
 
 class Soap
```
Comment (lukeraymonddowning) [inline src/Soap.php:9]: We can use `ForwardsCalls` on this.

**Review by lukeraymonddowning — CHANGES_REQUESTED** [review]

File: src/Providers/SoapServiceProvider.php (hunk 1 above)
Comment (nedwors) [reply src/Providers/SoapServiceProvider.php]: Sorted! Thanks

File: src/Soap.php (hunk 2 above)
Comment (nedwors) [reply src/Soap.php:10]: Implemented, thanks

**Pull request merged**